spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala

```diff
@@ -252,8 +252,12 @@ object BatchDownsampler extends StrictLogging with Instance {
                                     downsampledChunksToPersist: MMap[FiniteDuration, Iterator[ChunkSet]]): Unit = {
     // write all chunks to cassandra
     val writeFut = downsampledChunksToPersist.map { case (res, chunks) =>
+      // FIXME if listener below is not overridden to no-op, we get a SEGV because
```

Comment:
Indicate why the copy is made? What the copy is of?

Reply:
Copy is made because scala case classes are immutable. 
We are making a copy of the ChunkSet case class.

Reply:
I mean, add the comments to the code. ;-)